File: spec/models/tweet_spec.rb
require 'rails_helper'

describe Tweet do

	let(:tweet) { create :tweet }
	let(:tweet_hash) { attributes_for :tweet }
# before(:each) do
# 	@tweet = Tweet.create(
# 		username: 'this_guy',
# 		content: 'this guy is tweeting',
# 		stars: 2,
# 		latitude: 500,
# 		longitude: 300,
# 		url: "http://this_guy.com",
# 		twitter_id: 17,
# 		location: "Chicago, IL")
# end

	it 'should create a tweet' do
		expect(tweet).to be_a_kind_of(Tweet)
	end

	it 'validates tweet id is unique' do
		Tweet.create tweet_hash
		expect{Tweet.create(tweet_hash)}.to_not change{Tweet.all.count}
	end



end

File: jasmine/spec/TweetSpec.js
describe("Tweet", function() {
  var tweet;

  beforeEach(function() {
    tweet = new Tweet({
      id: 1,
      username: "bob",
      content: "tweet tweet",
      longitude: 72.45,
      latitude: 54.34,
      twitter_id: "9048750982374098234",
      location: "Chicago, IL",
      stars: 1
    });
  });

  it("is a Tweet", function() {
    console.log(tweet.created_at)
    expect(tweet instanceof Tweet).toBe(true);
  });

  it("should store the correct information", function() {
    expect(tweet.username).toBe("bob");
  });

  //   it("should be possible to resume", function() {
  //     player.resume();
  //     expect(player.isPlaying).toBeTruthy();
  //     expect(player.currentlyPlayingSong).toEqual(song);
  //   });
  // });

  // // demonstrates use of spies to intercept and test method calls
  // it("tells the current song if the user has made it a favorite", function() {
  //   spyOn(song, 'persistFavoriteStatus');

  //   player.play(song);
  //   player.makeFavorite();

  //   expect(song.persistFavoriteStatus).toHaveBeenCalledWith(true);
  // });

  // //demonstrates use of expected exceptions
  // describe("#resume", function() {
  //   it("should throw an exception if song is already playing", function() {
  //     player.play(song);

  //     expect(function() {
  //       player.resume();
  //     }).toThrowError("song is already playing");
  //   });
  // });
});
